File: src/app/models/group.py
from typing import List, Optional
from uuid import UUID
from pydantic import BaseModel, Field

class IdName(BaseModel):
    id: str
    name: str

class ViewGroupModel(BaseModel):
    id: int
    name: str
    type: str
    companyNames: Optional[List[str]] = None
    companyIds: Optional[List[int]] = None
    userIds: Optional[List[UUID]] = None
    active: bool

class CreateGroupModel(BaseModel):
    name: str
    type: str
    companyIds: List[int]
    userIds: List[UUID] = Field(default_factory=list)

class UpdateGroupModel(BaseModel):
    name: str
    type: str
    companyIds: List[int]
    userIds: List[UUID] = Field(default_factory=list)
    active: bool

File: src/client_app/src/features/admin/models/group.ts
export interface CreateGroupModel {
    name: string;
    type: string;
    companyIds: string[];
    userIds: string[];
}

export interface UpdateGroupModel extends CreateGroupModel {
    active: boolean;
}

export interface GroupModel {
    id: string;
    name: string;
    type: string;
    companyNames: string[];
    active: boolean;
}

export interface PaginatedGroupModel {
    items: GroupModel[];
    totalCount: number;
    totalPages: number;
}